mlrun/model_monitoring/controller.py

```diff
@@ -282,33 +282,35 @@ class MonitoringApplicationController:
 
     def __init__(
         self,
-        context: mlrun.run.MLClientCtx,
+        mlrun_context: mlrun.run.MLClientCtx,
         project: str,
     ):
         """
         Initialize Monitoring Application Processor object.
 
-        :param context:                     An MLRun context.
+        :param mlrun_context:               An MLRun context.
         :param project:                     Project name.
         """
-        self.context = context
+        self.context = mlrun_context
         self.project = project
         self.project_obj = mlrun.get_or_create_project(project)
 
-        context.logger.debug(f"Initializing {self.__class__.__name__}", project=project)
+        mlrun_context.logger.debug(
+            f"Initializing {self.__class__.__name__}", project=project
+        )
 
         self.db = mlrun.model_monitoring.get_model_endpoint_store(project=project)
 
         self._batch_window_generator = _BatchWindowGenerator(
-            batch_dict=context.parameters[
-                mm_constants.EventFieldType.BATCH_INTERVALS_DICT
-            ]
+            batch_dict=json.loads(
+                mlrun.get_secret_or_env(
+                    mm_constants.EventFieldType.BATCH_INTERVALS_DICT
+                )
+            )
         )
 
         # If provided, only model endpoints in that that list will be analyzed
-        self.model_endpoints = context.parameters.get(
-            mm_constants.EventFieldType.MODEL_ENDPOINTS, None
-        )
+        self.model_endpoints = None
```

Comment:
Not sure why do we need that `self._model_endpoints = None` here. How should the user provide this list? When the controller was a scheduled job I can understand that perhaps we would like to provide a list of model endpoints to process. However, now it's a real-time function. As I see it we have 4 options:
1 - use a flag/label on the tracked model endpoints
2 - provide optional model endpoints list through the incoming event to the controller handler
3 - store model endpoints list as an env/secret (seems unnecessary) 
4 - manage the processed model endpoints through the monitoring policy (once implemented)

Anyway, at the moment I think we can remove it from the controller code and implement it as a different PR with the selected approach.